rocky/reports/views/base.py

```diff
@@ -166,29 +165,135 @@ def get_report_types_from_choice(self) -> list[type[Report] | type[MultiReport]]
                 report_types.append(report)
             except ValueError:
                 error_message = _("Report type '%s' does not exist.") % report_type
-                messages.add_message(self.request, messages.ERROR, error_message)
+                messages.error(self.request, error_message)
         return report_types
 
-    def get_report_types(self) -> list[ReportType]:
+    @staticmethod
+    def get_report_types(reports: set[type[BaseReportType]]) -> list[dict[str, str]]:
         return [
             {
                 "id": report_type.id,
                 "name": report_type.name,
                 "description": report_type.description,
                 "label_style": report_type.label_style,
             }
-            for report_type in self.get_report_types_from_choice()
+            for report_type in reports
         ]
 
+
+class ReportPluginView(ReportOOIView, ReportTypeView, TemplateView):
+    """
+    This view shows the required and optional plugins.
+    Needs ReportTypeView to know which report type is selected to get their plugins.
+    The plugin ids will be collected and fetched form KAT-alogus.
+    The user is able to activate plugins they need for the scans.
+    The oois selection is also remembered by ReportOOIView.
+    """
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        try:
+            self.plugins, self.all_plugins_enabled = self.get_plugins()
+        except KATalogusError as error:
+            messages.error(self.request, error.message)
+            self.plugins = {}
+            self.all_plugins_enabled = {}
+
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context["created_at"] = datetime.now()
-        context["selected_oois"] = self.selected_oois
-        context["selected_report_types"] = self.selected_report_types
+
         context["plugins"] = self.plugins
-        context["oois"] = self.get_oois()
+        context["all_plugins_enabled"] = self.all_plugins_enabled
+        context["plugin_data"] = self.get_plugin_data()
+
         return context
 
+    def report_has_required_plugins(self) -> bool:
+        if self.plugins:
+            required_plugins = self.plugins["required"]
+            return required_plugins is not None
```

Comment:
`self.plugins["required"]` is set to the empty list if there are no plugins, so this will never be `None` as far as I can see...